src/realm/dictionary.cpp

```diff
@@ -40,6 +40,144 @@ DictionaryClusterTree::DictionaryClusterTree(ArrayParent* owner, DataType key_ty
 
 DictionaryClusterTree::~DictionaryClusterTree() {}
 
+Mixed DictionaryClusterTree::min(size_t* return_ndx) const
+{
+    Mixed m;
+    size_t ndx = realm::npos;
+    ArrayMixed leaf(m_alloc);
+    size_t start_ndx = 0;
+
+    traverse([&](const Cluster* cluster) {
+        size_t e = cluster->node_size();
+        cluster->init_leaf(s_values_col, &leaf);
+        for (size_t i = 0; i < e; i++) {
+            auto val = leaf.get(i);
+            if (val.is_null())
+                continue;
+
+            if (m.is_null() || val < m) {
+                m = val;
+                ndx = i + start_ndx;
+            }
+        }
+        start_ndx += e;
+        // Continue
+        return false;
+    });
+
+    if (return_ndx)
+        *return_ndx = ndx;
+
+    return m;
+}
+
+Mixed DictionaryClusterTree::max(size_t* return_ndx) const
+{
+    Mixed m;
+    size_t ndx = realm::npos;
+    ArrayMixed leaf(m_alloc);
+    size_t start_ndx = 0;
+
+    traverse([&](const Cluster* cluster) {
+        size_t e = cluster->node_size();
+        cluster->init_leaf(s_values_col, &leaf);
+        for (size_t i = 0; i < e; i++) {
+            auto val = leaf.get(i);
+            if (val.is_null())
+                continue;
+
+            if (m.is_null() || val > m) {
+                m = val;
+                ndx = i + start_ndx;
+            }
+        }
+        start_ndx += e;
+        // Continue
+        return false;
+    });
+
+    if (return_ndx)
+        *return_ndx = ndx;
+
+    return m;
+}
+
+Mixed DictionaryClusterTree::sum(size_t* return_cnt) const
+{
+    Mixed s;
+    size_t cnt = 0;
+    ArrayMixed leaf(m_alloc);
+
+    traverse([&](const Cluster* cluster) {
+        size_t e = cluster->node_size();
+        cluster->init_leaf(s_values_col, &leaf);
+        for (size_t i = 0; i < e; i++) {
+            auto val = leaf.get(i);
+            if (val.is_null())
+                continue;
+
+            cnt++;
+
+            if (s.is_null()) {
+                s = val;
+            }
+            else {
+                if (s.get_type() != val.get_type()) {
+                    throw std::runtime_error("Sum not supported");
+                }
+                switch (s.get_type()) {
+                    case type_Int:
+                        s = Mixed(s.get_int() + val.get_int());
+                        break;
+                    case type_Float:
+                        s = Mixed(s.get_float() + val.get_float());
+                        break;
+                    case type_Double:
+                        s = Mixed(s.get_double() + val.get_double());
+                        break;
+                    case type_Decimal:
+                        s = Mixed(s.get<Decimal128>() + val.get<Decimal128>());
+                        break;
+                    default:
+                        throw std::runtime_error("Average not supported");
```

Comment:
Consider changing the error messages to something more generic because this method is used for average and sum. Maybe even get fancy and print the type. (not blocking)